scripts/gen_attribute_completion_and_shape_inference_hpp.py

```diff
@@ -0,0 +1,322 @@
+       node.attribute_table.find("pads") == node.attribute_table.end()));
+    if(found != node.attribute_table.end()) {
+        auto output_shape = get<ints>(found->second);
+        /* [dim0_begin, dim1_begin, ... , dim0_end, dim1_end, ..., ...] */
+        ints pads(kernel_ndims*2, 0);
+        auto output_padding =
+            get<ints>(node.attribute_table.at("output_padding"));
+        auto strides = get<ints>(node.attribute_table.at("strides"));
+        auto input_profile = input_profile_table.at(input(0));
+        ints input_size(input_profile.dims().begin()+2,
+                        input_profile.dims().end());
+
+        for(int i = 0; i < kernel_ndims; ++i) {
+            auto total_padding = strides[i] * (input_size[i] - 1)
+                + output_padding[i] + kernel_shape[i] - output_shape[i];
+            pads[i] = total_padding - (total_padding/2);
+            pads[i+kernel_ndims] = (total_padding/2);
+        }
+
+        node.attribute_table["pads"] = pads;
+    }
+}
+'''))
+    code_list.append(make_completion_code("Elu", [("alpha", "float", "1.f")]))
+    code_list.append(
+        make_completion_code("FC", [], '''
+auto output_dims = ints({dims_of(input(0)).at(0), dims_of(input(1)).at(0)});
+add_variable_to_table(output(0), dtype_of(input(0)),
+    output_dims);
+'''))
+    code_list.append(
+        make_completion_code("Gemm", [
+            ("alpha", "float", "1.f"),
+            ("beta", "float", "1.f"),
+            ("transA", "int", "0"),
+            ("transB", "int", "0"),
+        ], '''
+auto a_dims = dims_of(input(0));
+assert(a_dims.size() == 2);
+if(transA) {
+    std::swap(a_dims.at(0), a_dims.at(1));
+}
+
+auto b_dims = dims_of(input(1));
+assert(b_dims.size() == 2);
+if(transB) {
+    std::swap(b_dims.at(0), b_dims.at(1));
+}
+
+auto output_dims = ints({a_dims.at(0), b_dims.at(1)});
+add_variable_to_table(output(0), dtype_of(input(0)), output_dims);
+'''))
+    code_list.append(
+        make_completion_code("LeakyRelu", [("alpha", "float", "0.01f")]))
+    code_list.append(
+        make_completion_code("LRN", [
+            ("alpha", "float", "0.0001f"),
+            ("beta", "float", "0.75f"),
+            ("bias", "float", "1.0f"),
+            ("size", "float", None),
+        ]))
+    code_list.append(
+        make_completion_code("MaxPool", [
+            ("kernel_shape", "ints", None),
+            ("pads", "ints", "ints(2*(ndims_of(input(0))-2), 0)"),
+            ("storage_order", "int", "0"),
+            ("strides", "ints", "ints(ndims_of(input(0))-2, 1)"), # WORKAROUND: None is correct # NOQA
+        ], '''
+add_variable_to_table(output(0), dtype_of(input(0)),
+    calc_2d_output_dims(
+        dims_of(input(0)), dims_of(input(0)).at(1),
+        kernel_shape, strides, pads));
+'''))
+    code_list.append(make_completion_code("Relu"))
+    code_list.append(make_completion_code("Softmax", [("axis", "int", "1")]))
+    code_list.append(make_completion_code("Sum"))
```

Comment:
Default `shape_inference_code` assumes an unary operator (it executes `assert(node.input_name_list.size() == 1)`), but `Sum` is a variadic operator.